src/aosm/azext_aosm/_configuration.py

```diff
@@ -59,9 +62,59 @@ def validate(self):
         if not self.version:
             raise ValidationError("version must be set.")
         if self.blob_sas_url and self.file_path:
-            raise ValidationError("Only one of file_path or blob_sas_url may be set.")
+            raise ValidationError(
+                "Only one of file_path or blob_sas_url may be set."
+            )
         if not (self.blob_sas_url or self.file_path):
-            raise ValidationError("One of file_path or sas_blob_url must be set.")
+            raise ValidationError(
+                "One of file_path or sas_blob_url must be set."
+            )
+
+
+@dataclass
+class VhdArtifactConfig(ArtifactConfig):
+    image_disk_size_GB: Optional[Union[str, int]] = None
+    image_os_state: Optional[str] = None
+    image_os_type: Optional[str] = None
+    image_hyper_v_generation: Optional[str] = None
+    image_api_version: Optional[str] = None
+
+    def __post_init__(self):
+        """
+        Convert parameters to the correct types.
+        """
+        if (
+            isinstance(self.image_disk_size_GB, str)
+            and self.image_disk_size_GB.isdigit()
+        ):
+            self.image_disk_size_GB = int(self.image_disk_size_GB)
+
+    @classmethod
+    def helptext(cls) -> "VhdArtifactConfig":
+        """
+        Build an object where each value is helptext for that field.
+        """
+        return VhdArtifactConfig(
+            image_disk_size_GB=(
+                "Optional. Specifies the size of empty data disks in gigabytes. "
+                "This value cannot be larger than 1023 GB."
+            ),
+            image_os_state=(
+                "Optional. The OS State. For managed images, use Generalized."
```

Comment:
What do we mean, "for managed images"?  Is there any other type of image when it comes to VNFs??

Is a link to https://learn.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries?tabs=azure-cli#generalized-and-specialized-images useful in the help here?

Reply:
I think you've taken the descriptions direct from e.g. https://dev.azure.com/msazuredev/AzureForOperators/_git/pez?path=/src/MultiaccessEdgeCompute/NetworkFunction/VNF/DurableTaskProcessor/Helpers/AzureCore/AzureCoreVnfTasks.cs&version=GBmain&line=1064&lineEnd=1065&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents 

We should do better.  If it is confusing there it'll be confusing in the CLI.

Reply:
The default is Generalized, so they don't have to set this for managed images, if that's the only sort we support.

Reply:
https://learn.microsoft.com/en-us/azure/virtual-machines/capture-image-resource - I've been trying to work out what a managed image is.  I _think_ it is an Azure image resource stored direct in a resource group (which is what we do in AOSM) rather than one stored in an Azure Compute Gallery (aka shared image gallery), which is the more modern way of doing things that allows for geo-replication.